crates/ruma-events/src/room/create.rs

```diff
@@ -34,15 +34,37 @@ pub struct CreateEventContent {
     /// A reference to the room this room replaces, if the previous room was upgraded.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub predecessor: Option<PreviousRoom>,
+
+    /// The room type that shall be set. This is currently used for spaces
+    #[serde(skip_serializing_if = "Option::is_none", rename = "type")]
+    pub room_type: Option<String>,
 }
 
 impl CreateEventContent {
     /// Creates a new `CreateEventContent` with the given creator.
     pub fn new(creator: UserId) -> Self {
-        Self { creator, federate: true, room_version: default_room_version_id(), predecessor: None }
+        Self {
+            creator,
+            federate: true,
+            room_version: default_room_version_id(),
+            predecessor: None,
+            room_type: None,
+        }
     }
 }
 
+/// An enum of possible room types
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[serde(untagged)]
+pub enum RoomType {
```

Comment:
This should be declared like all the other enums that are strings in the spec, see for example `RoomPreset` in `ruma_client_api::r0::room::create_room`.

Reply:
Fixed in 5fac97ae0653260c307b7779259fc95cb6f08d75